### PR TITLE
Removing softmax, closes #37

### LINKED ISSUE #37: Softmax layers on output of models
Hi, in the main.py in the training loop we are currently using `nn.CrossEntropyLoss()`, it should take as input the raw logits (not softmaxed probabilities) and targets - [https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html](CrossEntropyLoss) if I'm not mistaken.

### PR DESCRIPTION
**Pull request by hzavadil98**
Took the liberty to touch your models code to adress #37 , please fact-check the issue and if I'm not wrong review the PR.

### CODE REVIEW

File: tests/test_models.py
Comment (salomaestro) [inline tests/test_models.py:1]: Thanks for fixing the test also!

**Review by salomaestro — APPROVED** [review]
Thanks for fixing!
